lib/storage/bucket.js

```diff
@@ -203,6 +203,100 @@ function Bucket(storage, name) {
 }
 
 /**
+ * Combine mutliple files into one new file.
+ *
+ * @throws if a non-array is provided as sources argument.
+ * @throws if less than two sources are provided.
+ * @throws if no destination is provided.
+ * @throws if a content type cannot be determined for the destination file.
+ *
+ * @param {string[]|module:storage/file[]} sources - The source files that will
+ *     be combined.
+ * @param {string|module:storage/file} destination - The file you would like the
+ *     source files combined into.
+ * @param {function=} callback - The callback function.
+ *
+ * @example
+ * var 2013logs = bucket.file('2013-logs.txt');
+ * var 2014logs = bucket.file('2014-logs.txt');
+ *
+ * var allLogs = bucket.file('all-logs.txt');
+ *
+ * bucket.combine([
+ *   2013logs,
+ *   2014logs
+ * ], allLogs, function(err, newFile) {
+ *   // newFile === allLogs
+ * });
+ */
+Bucket.prototype.combine = function(sources, destination, callback) {
```

Comment:
I was thinking not. Another most common use cases > being an API passthrough case, which matches other method we have in the library (which don't allow complete access to the underlying API call). I think if we want to be 1:1 with the API, that'll require a new issue for discussion and careful examination of every single method we have to make sure it offers it.